src/overlays/actors/ovl_En_In/z_en_in.c

```diff
@@ -21,92 +45,1572 @@ const ActorInit En_In_InitVars = {
+            if (textId == 0x3473) {
+                func_808F35D8(this, globalCtx);
+            }
+            break;
+        case 0:
+            break;
+    }
+}
+
+void func_808F4108(EnIn* this, GlobalContext* globalCtx, u16 arg2) {
+    func_800B86C8(&this->actor, globalCtx, &this->unk4A4->actor);
+    this->actor.textId = 0;
+    this->unk4A4->actor.textId = arg2;
+    this->unk4A4->unk48C = 2;
+    this->unk48C = 0;
+}
+
+s32 func_808F4150(GlobalContext* globalCtx, EnIn* this, s32 arg2, MessageContext* msgCtx) {
+    EnIn* this2 = this;
+
+    if (msgCtx->choiceIndex == 0) {
+        func_8019F208();
+        if (gSaveContext.rupees >= globalCtx->msgCtx.unk1206C) {
+            func_801159EC(-globalCtx->msgCtx.unk1206C);
+            if (!(gSaveContext.weekEventReg[57] & 1)) {
+                func_808F4108(this, globalCtx, 0x3474);
+            } else if (this->unk4AC & 8) {
+                func_808F4108(this, globalCtx, 0x3475);
+            } else {
+                func_800E8EA0(globalCtx, &this->actor, 0x3475);
+            }
+        } else {
+            play_sound(0x4806);
+            func_800E8EA0(globalCtx, &this->actor, 0x3473);
+        }
+    } else {
+        func_8019F230();
+        func_800E8EA0(globalCtx, &this->actor, 0x3472);
+    }
+    return 0;
+}
+
+s32 func_808F4270(GlobalContext* globalCtx, EnIn* this, s32 arg2, MessageContext* msgCtx, s32 arg4) {
+    s32 pad;
+    s32 fee = globalCtx->msgCtx.unk1206C != 0xFFFF ? globalCtx->msgCtx.unk1206C : 10;
+
+    if (msgCtx->choiceIndex == 0) {
+        func_8019F208();
+        if (gSaveContext.rupees >= fee) {
+            func_801159EC(-fee);
+            if (!(gSaveContext.weekEventReg[57] & 1)) {
+                if (arg4 != 0) {
+                    func_800E8EA0(globalCtx, &this->actor, 0x3474);
+                } else {
+                    func_808F4108(this, globalCtx, 0x3474);
+                }
+            } else {
+                if (arg4 != 0) {
+                    func_800E8EA0(globalCtx, &this->actor, 0x3475);
+                } else {
+                    func_808F4108(this, globalCtx, 0x3475);
+                }
+            }
+        } else {
+            play_sound(0x4806);
+            if (arg4 != 0) {
+                func_800E8EA0(globalCtx, &this->actor, 0x3473);
+            } else {
+                func_808F4108(this, globalCtx, 0x3473);
+            }
+        }
+    } else {
+        func_8019F230();
+        func_800E8EA0(globalCtx, &this->actor, 0x3472);
+    }
+    return 0;
+}
+
+s32 func_808F43E0(EnIn* this) {
+    this->unk48C = 0;
+    this->actor.textId = 0;
+    SET_FLAGS_FINISH_RACE;
+    return 0;
+}
+
+s32 func_808F4414(GlobalContext* globalCtx, EnIn* this, s32 arg2) {
+    u16 textId = this->actor.textId;
+    s32 ret = 1;
+    MessageContext* msgCtx = &globalCtx->msgCtx;
+    s32 pad[2];
+
+    switch (textId) {
+        case 0x34AF:
+            func_800E8EA0(globalCtx, &this->actor, 0x34B0);
+            ret = 0;
```

Comment:
```suggestion
            ret = false;
```
Looks like you could change all returns to `true` and `false` for this function